client/js/components/home.jsx

```diff
@@ -26,23 +26,33 @@ class Home extends React.Component {
     }
   }
 
-  checkItem(child, e){
-    let itemChecked = this.state.itemChecked;
-    itemChecked[child.id] = e.target.checked;
+  checkItem(bank, value){
+    let map = {[bank.id]: value};
+    if(!value){
+      this.resetHierarchy(bank, map);
+    }
+    let itemChecked = Object.assign({}, this.state.itemChecked, map);
     this.setState({itemChecked});
   }
 
-  renderItem(child){
+  resetHierarchy(bank, map){
+    map[bank.id] = false;
+    _.forEach(bank.childNodes, (bc)=>{
+      this.resetHierarchy(bc, map);
+    });
+  }
+
+  renderItem(bank){
     let itemClass = "c-filter__item";
-    if(!_.isUndefined(child.children)){
+    if(!_.isUndefined(bank.childNodes)){
```

Comment:
I think this will always be true.  Also, I can't figure out what the "c-filter__item--dropdown" style class is for.